# Gene Model Prediction, Annotation, & Analyses

## Ab initio gene prediction:  
1) Prior to gene prediction, repetitive sequences were soft-masked used RepeatModeler (v2.0.2) and RepeatMasker (v4.1.2). See 1_soft-masking.sh  
  
2) RNAseq datasets (listed below) were downloaded from NCBI SRA and mapped to the genome with HISAT2 (v2.2.1) to use as evidence with BRAKER. See 2_hisat2-rna-mapping.sh
- [PRJNA637102](https://www.ncbi.nlm.nih.gov/bioproject/PRJNA637102) (Temperature and pCO2 effects on red sea urchin embryo gene expression)
- [PRJNA531463](https://www.ncbi.nlm.nih.gov/bioproject/PRJNA531463) (*M. franciscanus* early developmental transcriptome)
- [PRJNA272924](https://www.ncbi.nlm.nih.gov/bioproject/PRJNA272924) (Transcriptomic responses of sea urchin larvae to Ocean Acidification)
- [PRJNA562282](https://www.ncbi.nlm.nih.gov/bioproject/PRJNA562282) (Age-related gene expression in tissues of the long-lived red sea urchin)
  
3) Gene model prediction using BRAKER (v2.1.6) with RNAseq data as evidence.  See 3_braker2.sh  
  
4) Gene model QC. See 4_geneModel-QC.sh 
- note that genes of particular interest were manually checked and corrected if necessary.  

## Annotation:
QC'ed gene models were annotated with SwissProt (2021-04 database) and RefSeq. See Annotation.sh  
  

## Gene Density Plots
Gene density plots were generated by Kate Castellano (see Gene_Density)
